### DISCUSSION TITLE
RP-Initiated Logout - POST Method - No Redirection

### DISCUSSION THREAD

**Discussion by Sp4m4** [keycloak/keycloak#14566] [Q&A]
Hello, 

I'm not sure  but it seems that Keycloack is not fully compliant with **RP-Initiated Logout** as described in [OpenID Connect RP-Initiated Logout 1.0](https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RPLogout).

When using POST method to diconnect user with the optionnal parameter **post_logout_redirect_uri**, this one is not taken. 
User is well disconnected but not redirected to **post_logout_redirect_uri**  instead of i get **204 status Code** ( the request was successful but the client does not need to leave the current page).

It seems that keycloak can do this redirection (https://www.keycloak.org/docs/latest/server_admin/#rp-initiated-logout)  although the method to use is not detailed. In doubt I suppose that it supports the 2 (GET and POST) 

Am i wrong ? 

thanks a lot

**Comment by sschu** [comment]
Did you configure the post_logout_redirect_uri for the client? Did you also provide client_id or id_token_hint?

**Reply by Sp4m4** [reply]
Client configuration has post_logout_redirect_ur parameter filled with https://myurl/oidc/redirect
Client is defined as confidential.

To request the logout endpoint through POSTMAN : 

Authorization set on basic with Client_ID/Secret
Body content as x-www-form-urlencoded
Parameters in body : 

- Refresh_token  : <refresh_token>
- post_logout_redirect_uri : https://myurl/oidc/redirect

id_token_hint parameter is for ID_Token  not refresh token. 

I got in return Status 204 and the session to the RT  is well disconnected because i am no more able to call UserInfo with the Access_Token

If I change the redirection URL (which must be strictly identical to the one configured) I get the status 204 and I am well disconnected. 
I have the same result by not setting the post_logout_redirect_uri. 
So I wonder if the POST method is fully implemented to take into account the optional parameters. 

I saw that RP-initiated logout was implemented in Keycloack recently, so, could the implementation be not complete ? 
https://github.com/keycloak/keycloak/issues/10885

**Reply by sschu** [reply]
Could be. Maybe have a look at the code to check if its actually supposed to do what you want?

**Comment by Sp4m4** [comment]
Unfortunately, as Identity Service Provider i have only POSTMAN to check  the steps of the OIDC operation. 
This behvaior was reported to me by the service providers using my IDP and i have the same Code result (204).

The GET Deconnexion returns the expected code 200

**Comment by Sp4m4** [comment]
does anybody has this issue ?
